### ISSUE TITLE
purge-cluster playbook fails if docker image is missing

### ISSUE THREAD

**Issue by Hobadee** [ceph/ceph-ansible#5000]
### What happened:
When using the `purge-cluster.yml` script, it fails on task "remove node-exporter image".  This didn't happen on v4.0.5, but does occur on v4.0.12.  This occurs on all cluster nodes, but for simplicity I will just use a single node as an example.

This is occurring because the image which is trying to be removed doesn't exist:
```
TASK [remove node-exporter image] *** 
task path: /Users/erick/src/ansible/ceph-ansible/purge-cluster.yml:155 ***
fatal: [stor01]: FAILED! => changed=true
  cmd:
  - docker
  - rmi
  - prom/node-exporter:v0.17.0
  delta: '0:00:00.054147'
  end: '2020-01-28 09:56:58.553215'
  msg: non-zero return code
  rc: 1
  start: '2020-01-28 09:56:58.499068'
  stderr: 'Error response from daemon: No such image: prom/node-exporter:v0.17.0'
  stderr_lines: <omitted>
  stdout: ''
  stdout_lines: <omitted>
```

Here is a list of docker images actually on the system:
```
stor01$ sudo docker images
REPOSITORY                     TAG                 IMAGE ID            CREATED             SIZE
docker.io/prom/node-exporter   latest              e5a616e4b9cf        7 months ago        22.9 MB
docker.io/grafana/grafana      5.2.4               920eb69ade2a        16 months ago       245 MB
```

Additionally, trying to manually remove the image using a similar command line to what ceph-ansible is generating, but with the appropriate image name, fails:
```
stor01$ sudo docker rmi docker.io/prom/node-exporter:latest
Error response from daemon: conflict: unable to remove repository reference "docker.io/prom/node-exporter:latest" (must force) - container 4725f4778c2c is using its referenced image e5a616e4b9cf
```

To actually remove the image, the following command works:
```
stor01$ sudo docker rmi -f docker.io/prom/node-exporter:latest
Untagged: docker.io/prom/node-exporter:latest
Untagged: docker.io/prom/node-exporter@sha256:a2f29256e53cc3e0b64d7a472512600b2e9410347d53cdc85b49f659c17e02ee
Deleted: sha256:e5a616e4b9cf68dfcad7782b78e118be4310022e874d52da85c55923fb615f87

stor01$ sudo docker images
REPOSITORY                  TAG                 IMAGE ID            CREATED             SIZE
docker.io/grafana/grafana   5.2.4               920eb69ade2a        16 months ago       245 MB
```

### What you expected to happen:
`purge-cluster.yml` should run successfully, purging all the cluster data

### How to reproduce it (minimal and precise):
1. Install a Ceph cluster using ceph-ansible
1. Purge the cluster using ceph-ansible's purge-cluster.yml playbook

### Workaround
You can workaround this issue by skipping the task which purges the images.  This can be done by adding `--skip-tags=remove_img` to the `ansible-playbook` command line.  Note that doing this will keep the docker images on your machines and not purge said docker images.

### Share your group_vars files, inventory
all.yml
```
---
dummy:
fetch_directory: fetch/
configure_firewall: True
ceph_mon_firewall_zone: public
ceph_mgr_firewall_zone: public
ceph_osd_firewall_zone: public
ceph_iscsi_firewall_zone: public
ceph_dashboard_firewall_zone: public
ceph_conf_local: true
centos_package_dependencies:
  - epel-release
  - libselinux-python
ceph_origin: repository
ceph_repository: community
ceph_stable_release: nautilus
cephx: true
monitor_address_block: 10.7.11.0/24
ip_version: ipv4
public_network: 10.7.11.0/24
cluster_network: 10.7.12.0/24
ceph_conf_overrides:
  osd:
    osd heartbeat grace: 20
    osd heartbeat interval: 5
ceph_tcmalloc_max_total_thread_cache: 0
dashboard_admin_password: <REDACTED>
dashboard_rgw_api_no_ssl_verify: true
```

### OS (e.g. from /etc/os-release):
```
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```
### Kernel (e.g. `uname -a`)
`Linux stor01 3.10.0-957.el7.x86_64 #1 SMP Thu Nov 8 23:39:32 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`
### Docker version if applicable (e.g. `docker version`)
`Docker version 1.13.1, build 7f2769b/1.13.1`
### Ansible version (e.g. `ansible-playbook --version`)
```
ansible-playbook 2.8.2
  config file = /Users/erick/src/ansible/ceph-ansible/ansible.cfg
  configured module search path = [u'/Users/erick/src/ansible/ceph-ansible/library']
  ansible python module location = /usr/local/lib/python2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 2.7.17 (default, Oct 24 2019, 12:57:47) [GCC 4.2.1 Compatible Apple LLVM 11.0.0 (clang-1100.0.33.8)]
```
### ceph-ansible version (e.g. `git head or tag or stable branch`)
`v4.0.12`
### Ceph version (e.g. `ceph -v`)
`ceph version 14.2.6 (f0aa067ac7a02ee46ea48aa26c6e298b5ea272e9) nautilus (stable)`

**Comment by mesenger** [comment]
Hi, I have very similar or maybe even the same issue. 

ceph-ansible version: 4.0.14

Ansible version:
ansible-playbook 2.8.12
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.6/site-packages/ansible
  executable location = /bin/ansible-playbook
  python version = 3.6.8 (default, Dec  5 2019, 15:45:45) [GCC 8.3.1 20191121 (Red Hat 8.3.1-5)]

What I noticed, is that those wrong image names are defined in defaults file of ceph-defaults role:

roles/ceph-defaults/defaults/main.yml:
node_exporter_container_image: "prom/node-exporter:v0.17.0"
grafana_container_image: "grafana/grafana:5.2.4"
prometheus_container_image: "prom/prometheus:v2.7.2"
alertmanager_container_image: "prom/alertmanager:v0.16.2"

**Comment by dsavineau** [comment]
This has been fixed in v4.0.15 via https://github.com/ceph/ceph-ansible/commit/dcd4b2955a8c06bb44fd2b2c5e7b37c29e8ab2f9
